src/main/java/com/coreos/jetcd/internal/impl/MaintenanceImpl.java

```diff
@@ -133,65 +97,27 @@
    */
   @Override
   public CompletableFuture<DefragmentResponse> defragmentMember(String endpoint) {
-    Optional<Pair<ManagedChannel, Optional<String>>> pairOptional = Optional.empty();
-    try {
-      pairOptional = Optional.of(this.dialFunction.dial(endpoint));
-      Pair<ManagedChannel, Optional<String>> pair = pairOptional.get();
-      ManagedChannel channel = pair.getKey();
-      Optional<String> token = pair.getValue();
-      MaintenanceGrpc.MaintenanceFutureStub stub = ClientUtil
-          .configureStub(MaintenanceGrpc.newFutureStub(channel), token);
-      ListenableFuture<DefragmentResponse> defragmentResponseListenableFuture = stub
-          .defragment(DefragmentRequest.getDefaultInstance());
-
-      // close channel when defragmentResponseListenableFuture completes.
-      defragmentResponseListenableFuture
-          .addListener(() -> channel.shutdownNow(), this.executorService);
-      return FutureConverter.toCompletableFuture(defragmentResponseListenableFuture);
-    } catch (Exception e) {
-      pairOptional.ifPresent((pair) -> pair.getKey().shutdownNow());
-      throw new RuntimeException("defragmentMember encounters error", e.getCause());
-    }
+    return toCompletableFuture(this.stub.defragment(DefragmentRequest.getDefaultInstance()));
```

Comment:
`defragmentMember` takes an endpoint. The request must send to that specific endpoint in order for this call to be correct. that's why I had a dialFunction that create a channel then use that channel to build a stub and send the request to.

```java
pairOptional = Optional.of(this.dialFunction.dial(endpoint));		
 // create ManagedChannel just for the endpoint
Pair<ManagedChannel, Optional<String>> pair = pairOptional.get(); 
ManagedChannel channel = pair.getKey();		
Optional<String> token = pair.getValue();
// build stub	
MaintenanceGrpc.MaintenanceFutureStub stub = ClientUtil		
   .configureStub(MaintenanceGrpc.newFutureStub(channel), token);	
// send defrag req to the node corresponding to the endpoint	
ListenableFuture<DefragmentResponse> defragmentResponseListenableFuture = stub		
.defragment(DefragmentRequest.getDefaultInstance());
```